### PR TITLE
3D Tiles in 2D/CV

### PR DESCRIPTION
**Pull request by bagnell**
Add 3D Tiles support for 2D and Columbus View.

The majority of this change was for models using RTC or are positioned in ECF (model matrix is identity). I'll open a PR for those changes in master. @lilleyse Can you create two simple glTF models from the tilesets you made for me for testing?

- [x] Add tests.

### CODE REVIEW

**Comment by lilleyse** [comment]
Contains local, ecef, quantized, and rtc:

[models.zip](https://github.com/AnalyticalGraphicsInc/cesium/files/715611/models.zip)

**Comment by pjcozzi** [comment]
>  Add tests.

There's also some new selection/visibility failures in this branch.

File: Apps/Sandcastle/gallery/3D Tiles.html [hunk 1]
```diff
@@ -30,7 +30,6 @@
 //Sandcastle_Begin
 
 var viewer = new Cesium.Viewer('cesiumContainer', {
-    scene3DOnly : true,
```
Comment (pjcozzi) [inline Apps/Sandcastle/gallery/3D Tiles.html]: Remove this from the other 3D Tiles Sandcastle examples.

**Comment by pjcozzi** [comment]
* Run http://localhost:8080/Apps/Sandcastle/index.html?src=3D%20Tiles.html&label=Showcases
* Switch to 2D
* Switch to 3D
* In the console:
```
Uncaught DeveloperError: normalized result is not a number
Error
    at new DeveloperError (http://localhost:8080/Source/Core/DeveloperError.js:44:19)
    at Function.Cartesian3.normalize (http://localhost:8080/Source/Core/Cartesian3.js:422:19)
    at PerspectiveOffCenterFrustum.computeCullingVolume (http://localhost:8080/Source/Scene/PerspectiveOffCenterFrustum.js:221:20)
    at getPickPerspectiveCullingVolume (http://localhost:8080/Source/Scene/Scene.js:2544:26)
    at getPickCullingVolume (http://localhost:8080/Source/Scene/Scene.js:2552:16)
    at Scene.pick (http://localhost:8080/Source/Scene/Scene.js:2593:36)
    at <anonymous>:171:31
    at handleMouseMove (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:253:13)
    at handlePointerMove (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:649:13)
    at HTMLCanvasElement.listener (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:68:13) (on line 422 of http://localhost:8080/Source/Core/Cartesian3.js)
[Color]: Visited: 1, Selected: 1, Commands: 0 | Requests: 0, Attempted: 0, Processing: 0, Content Ready: 5, Total: 5 | Tiles styled: 0, Features styled: 0
Uncaught DeveloperError: normalized result is not a number
Error
    at new DeveloperError (http://localhost:8080/Source/Core/DeveloperError.js:44:19)
    at Function.Cartesian3.normalize (http://localhost:8080/Source/Core/Cartesian3.js:422:19)
    at PerspectiveOffCenterFrustum.computeCullingVolume (http://localhost:8080/Source/Scene/PerspectiveOffCenterFrustum.js:221:20)
    at getPickPerspectiveCullingVolume (http://localhost:8080/Source/Scene/Scene.js:2544:26)
    at getPickCullingVolume (http://localhost:8080/Source/Scene/Scene.js:2552:16)
    at Scene.pick (http://localhost:8080/Source/Scene/Scene.js:2593:36)
    at <anonymous>:171:31
    at handleMouseMove (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:253:13)
    at handlePointerMove (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:649:13)
    at HTMLCanvasElement.listener (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:68:13) (on line 422 of http://localhost:8080/Source/Core/Cartesian3.js)
Uncaught DeveloperError: normalized result is not a number
Error
    at new DeveloperError (http://localhost:8080/Source/Core/DeveloperError.js:44:19)
    at Function.Cartesian3.normalize (http://localhost:8080/Source/Core/Cartesian3.js:422:19)
    at PerspectiveOffCenterFrustum.computeCullingVolume (http://localhost:8080/Source/Scene/PerspectiveOffCenterFrustum.js:221:20)
    at getPickPerspectiveCullingVolume (http://localhost:8080/Source/Scene/Scene.js:2544:26)
    at getPickCullingVolume (http://localhost:8080/Source/Scene/Scene.js:2552:16)
    at Scene.pick (http://localhost:8080/Source/Scene/Scene.js:2593:36)
    at <anonymous>:171:31
    at handleMouseMove (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:253:13)
    at handlePointerMove (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:649:13)
    at HTMLCanvasElement.listener (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:68:13) (on line 422 of http://localhost:8080/Source/Core/Cartesian3.js)
Uncaught DeveloperError: normalized result is not a number
Error
    at new DeveloperError (http://localhost:8080/Source/Core/DeveloperError.js:44:19)
    at Function.Cartesian3.normalize (http://localhost:8080/Source/Core/Cartesian3.js:422:19)
    at PerspectiveOffCenterFrustum.computeCullingVolume (http://localhost:8080/Source/Scene/PerspectiveOffCenterFrustum.js:221:20)
    at getPickPerspectiveCullingVolume (http://localhost:8080/Source/Scene/Scene.js:2544:26)
    at getPickCullingVolume (http://localhost:8080/Source/Scene/Scene.js:2552:16)
    at Scene.pick (http://localhost:8080/Source/Scene/Scene.js:2593:36)
    at <anonymous>:171:31
    at handleMouseMove (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:253:13)
    at handlePointerMove (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:649:13)
    at HTMLCanvasElement.listener (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:68:13) (on line 422 of http://localhost:8080/Source/Core/Cartesian3.js)
```

**Comment by pjcozzi** [comment]
* Run http://localhost:8080/Apps/Sandcastle/index.html?src=3D%20Tiles.html&label=Showcases
* Switch to CV
* Load `Translucent` tileset
* Then load the top `Tileset` tileset
* Get this exception:

image

**Comment by pjcozzi** [comment]
Part of #3241

**Comment by pjcozzi** [comment]
The 3D Tiles Batch Table Hierarchy example works in 2D, but not in CV.

**Comment by pjcozzi** [comment]
The 3D Tiles Point Cloud Styling example only works in 3D; the tileset does not show up in 2D or CV.

**Comment by pjcozzi** [comment]
@pmconne - keep an eye on progress here for 3D Tiles in 2D and Columbus View.

File: Source/Scene/Cesium3DTileset.js [hunk 2]
```diff
@@ -1053,25 +1053,35 @@ define([
...
-            error -= dynamicError;
+            var pixelSize = Math.max(frustum.top - frustum.bottom, frustum.right - frustum.left) / Math.max(width, height);
+            error = geometricError / pixelSize;
+        } else {
+            // Avoid divide by zero when viewer is inside the tile
+            var distance = Math.max(tile.distanceToCamera, CesiumMath.EPSILON7);
+            var sseDenominator = camera.frustum.sseDenominator;
+            error = (geometricError * height) / (distance * sseDenominator);
+
+            if (tileset.dynamicScreenSpaceError) {
```
Comment (pjcozzi) [inline Source/Scene/Cesium3DTileset.js]: Can we still account for this in CV?  What about 2D?

**Comment by pjcozzi** [comment]
@bagnell it would be good to test this with a massive tileset and make sure 2D uses the SSE to not load too many tiles.

**Comment by pjcozzi** [comment]
Just those comments, good start @bagnell.

**Comment by lilleyse** [comment]
Here are some updated models: 
[gltfs.zip](https://github.com/AnalyticalGraphicsInc/cesium/files/727725/gltfs.zip)

File: Source/Scene/Cesium3DTileset.js (hunk 2 above)
Comment (bagnell) [inline Source/Scene/Cesium3DTileset.js]: I don't think it would make sense in 2D. Maybe CV, but we don't have fog in CV either.

**Comment by bagnell** [comment]
This is ready for another look. I still need to add tests, but all of the issues mentioned should be fixed.

**Comment by pjcozzi** [comment]
@bagnell is this 100% ready since tests where added in 1f707177c96995527dabfd7af3ad04f843440d6f?

**Comment by bagnell** [comment]
Yes, the rest of the tests are in #4922.

**Comment by pjcozzi** [comment]
In the 3D Tiles Batch Table Hierarchy example:

* Switch to 2D then back to 3D.
* The follow exception shows up in the console:

```
Uncaught DeveloperError: normalized result is not a number
Error
    at new DeveloperError (http://localhost:8080/Source/Core/DeveloperError.js:44:19)
    at Function.Cartesian3.normalize (http://localhost:8080/Source/Core/Cartesian3.js:422:19)
    at PerspectiveOffCenterFrustum.computeCullingVolume (http://localhost:8080/Source/Scene/PerspectiveOffCenterFrustum.js:221:20)
    at getPickPerspectiveCullingVolume (http://localhost:8080/Source/Scene/Scene.js:2544:26)
    at getPickCullingVolume (http://localhost:8080/Source/Scene/Scene.js:2552:16)
    at Scene.pick (http://localhost:8080/Source/Scene/Scene.js:2593:36)
    at <anonymous>:211:31
    at handleMouseMove (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:253:13)
    at handlePointerMove (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:649:13)
    at HTMLCanvasElement.listener (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:68:13) (on line 422 of http://localhost:8080/Source/Core/Cartesian3.js)
Uncaught DeveloperError: normalized result is not a number
Error
    at new DeveloperError (http://localhost:8080/Source/Core/DeveloperError.js:44:19)
    at Function.Cartesian3.normalize (http://localhost:8080/Source/Core/Cartesian3.js:422:19)
    at PerspectiveOffCenterFrustum.computeCullingVolume (http://localhost:8080/Source/Scene/PerspectiveOffCenterFrustum.js:221:20)
    at getPickPerspectiveCullingVolume (http://localhost:8080/Source/Scene/Scene.js:2544:26)
    at getPickCullingVolume (http://localhost:8080/Source/Scene/Scene.js:2552:16)
    at Scene.pick (http://localhost:8080/Source/Scene/Scene.js:2593:36)
    at <anonymous>:211:31
    at handleMouseMove (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:253:13)
    at handlePointerMove (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:649:13)
    at HTMLCanvasElement.listener (http://localhost:8080/Source/Core/ScreenSpaceEventHandler.js:68:13) (on line 422 of http://localhost:8080/Source/Core/Cartesian3.js)
```

**Comment by pjcozzi** [comment]
In the 3D Tiles Sandcastle example, the "Instanced" tileset does not show up in 2D and CV.

**Comment by bagnell** [comment]
@pjcozzi This is ready.

**Comment by pjcozzi** [comment]
Looks good!

**Pull request merged**